presto-hive/src/main/java/io/prestosql/plugin/hive/procedure/DropStatsProcedure.java

```diff
@@ -86,20 +89,27 @@ public Procedure get()
                 DROP_STATS.bindTo(this));
     }
 
-    public void dropStats(ConnectorSession session, String schema, String table, List<?> partitionValues)
+    public void dropStats(ConnectorSession session, ConnectorAccessControl accessControl, String schema, String table, List<?> partitionValues)
     {
         try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
-            doDropStats(session, schema, table, partitionValues);
+            doDropStats(session, accessControl, schema, table, partitionValues);
         }
     }
 
-    private void doDropStats(ConnectorSession session, String schema, String table, List<?> partitionValues)
+    private void doDropStats(ConnectorSession session, ConnectorAccessControl accessControl, String schema, String table, List<?> partitionValues)
     {
         TransactionalMetadata hiveMetadata = hiveMetadataFactory.create();
         HiveTableHandle handle = (HiveTableHandle) hiveMetadata.getTableHandle(session, new SchemaTableName(schema, table));
         if (handle == null) {
             throw new PrestoException(INVALID_PROCEDURE_ARGUMENT, format("Table '%s' does not exist", new SchemaTableName(schema, table)));
         }
+
+        try {
+            accessControl.checkCanDeleteFromTable(null, new SchemaTableName(schema, table));
```

Comment:
I am not sure if it is a good check that is called here. You don't delete data, you modify metadata. In `io.prestosql.sql.analyzer.StatementAnalyzer.Visitor#visitAnalyze` we check privilege for insert which is also weird.

I think we should have dedicated method for modifying statistics.

@electrum Any thoughts?

Reply:
Having a dedicated check for statistics makes sense, given that we have explicit checks for other operations, and statistics modification is part of the SPI for `ANALYZE`.

Reply:
It adds additional api call for implementors of security checks. I think  we should defer adding a dedicated check until we have a use-case at hand, which may be never.